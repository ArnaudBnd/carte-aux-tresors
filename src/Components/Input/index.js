import React, { useState } from 'react'
import { MapInit } from '../Map/index.js' 

export function InputFolder() {
  const hiddenFileInput = React.useRef(null)
  const [mapCoordonnes, setMapCoordonnes] = useState([])
  
  const handleClick = event => {
    hiddenFileInput.current.click()
  }

  const handleChange = event => {
    let fileData = new FileReader()
    const myFile = event.target.files[0]

    fileData.onload = (fileLoadedEvent) => {
      const textFromFileLoaded = fileLoadedEvent.target.result
      const textSplit = textFromFileLoaded.split(/\n/)
      const suppressComment = []

      textSplit.map(elmt => {
        if(elmt.search('#') === 0) {
          return
        } else {
          suppressComment.push(elmt.toUpperCase().split(' - '))
        }
      })

      setMapCoordonnes(suppressComment)
    }

    fileData.readAsText(myFile, "UTF-8")
  }

  return (
    <div>
      <button onClick={handleClick}>
        Upload a file
      </button>
      <input type="file"
             ref={hiddenFileInput}
             onChange={handleChange}
             style={{display: 'none'}}
      /> 
      <MapInit coordonnes={mapCoordonnes} />
    </div>
  )
}