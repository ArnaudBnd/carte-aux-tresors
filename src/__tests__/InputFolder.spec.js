import React from 'react'
import { InputFolder } from '../Components/Input/index'
import Enzyme from 'enzyme'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() });

describe("InputFolder", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<InputFolder />)
  })

  it('renders', () => {
   expect(wrapper).not.toBeNull()
  }) 

  it('renders App component', () => {
   expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('contains a button', () => {
    const button = <button> Upload a file </button>
    expect(wrapper.containsMatchingElement(button)).toBe(true)
  })

  it('calls handleClick when upload file is clicked', () => {
    wrapper.find('button').simulate('onClick')
  })

  it('contains a input', () => {
    const input = <input></input>
    expect(wrapper.containsMatchingElement(input)).toBe(true)
  })

  it('contains a props type name file', () => {
    const input = wrapper.find('input')
    expect(input.props().type).toBe('file')
  })

  it('calls handleChange when upload file is selected', () => {
    wrapper.find('input').simulate('onChange')
  })
})