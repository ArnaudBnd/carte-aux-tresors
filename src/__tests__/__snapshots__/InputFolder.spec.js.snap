// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFolder renders App component 1`] = `
<div>
  <button
    id="buttonDownloadFile"
    onClick={[Function]}
  >
    Upload a file
  </button>
  <input
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
  <MapInit
    coordonnes={Array []}
  />
</div>
`;
